cmd/traffic/cmd/manager/service.go

```diff
@@ -565,6 +566,72 @@ func (m *Manager) WatchLookupHost(session *rpc.SessionInfo, stream rpc.Manager_W
 	}
 }
 
+// GetLogs acquires the logs for the traffic-manager and/or traffic-agents specified by the
+// GetLogsRequest and returns them to the caller
+func (m *Manager) GetLogs(ctx context.Context, request *rpc.GetLogsRequest) (*rpc.LogsResponse, error) {
+	resp := &rpc.LogsResponse{
+		PodLogs: make(map[string]string),
+	}
+	var errMsg string
+	client := managerutil.GetKatesClient(ctx)
+
+	// getPodLogs is a helper function for getting the logs from the container
+	// of a given pod. If we are unable to get a log for a given pod, we will
+	// instead return the error in the map, instead of the log, so that:
+	// - one failure doesn't prevent us from getting logs from other pods
+	// - it is easy to figure out why gettings logs for a given pod failed
+	getPodLogs := func(pods []*kates.Pod, container string) {
+		for _, pod := range pods {
```

Comment:
should this be parallelized? it seems like if there's a lot of agents this could take a while